static/js/components/subject_page/main_pane.tsx

```diff
@@ -63,9 +76,25 @@ export const SubjectPageMainPane = memo(function SubjectPageMainPane(
 
   useEffect(() => {
     // re-fetch geojson data if props change
-    fetchGeoJsonData(props.place, enclosedPlaceType, props.parentPlaces).then(
-      (geoJsonData) => {
-        setGeoJsonData(geoJsonData);
+    const childrenGeoJsonPromise = fetchGeoJsonData(
+      props.place,
+      enclosedPlaceType,
+      props.parentPlaces
+    );
+    let placeGeoJsonProp = "";
+    for (const type of props.place.types) {
+      if (type in PLACE_TYPE_GEOJSON_PROP) {
+        placeGeoJsonProp = PLACE_TYPE_GEOJSON_PROP[type];
+        break;
+      }
+    }
+    const placeGeoJsonPromise = fetchEntityGeoJson(
```

Comment:
we might want to rename the function (and route)

Reply:
since this function/route fetches geojson for places or events or any list of "entities", I went with a general name. What would you suggest?

Reply:
done - updated to NodeGeoJson